src/sql/src/func.rs

```diff
@@ -2197,7 +2197,7 @@ pub static PG_CATALOG_BUILTINS: Lazy<BTreeMap<&'static str, Func>> = Lazy::new(|
                     func: VariadicFunc::RangeCreate { elem_type: ScalarType::Numeric { max_scale: None } },
                     exprs
                 })
-            }) => ScalarType::Range { element_type: Box::new(ScalarType::Int32)}, 3844;
+            }) =>  ScalarType::Range { element_type: Box::new(ScalarType::Numeric { max_scale: None })}, 3844;
```

Comment:
Was this an existing bug? Are there any backwards compatibility issues?

Reply:
This is an existing bug and not any meaningful compatibility issues because the expression still evaluated to the right type, its return type in the catalog was just wrong; if anyone depended on that, they would not have been able to use this function without running into an error.